Integrating With Other Libraries guide translation (#124)
reactjs/de.reactjs.org: content/docs/integrating-with-other-libraries.md

---
id: integrating-with-other-libraries
title: Integration mit anderen Bibliotheken
permalink: docs/integrating-with-other-libraries.html
---

> React kann in jeder Webanwendung verwendet werden. Es kann in andere Anwendungen eingebettet werden, und mit ein wenig Sorgfalt können andere Anwendungen in React eingebettet werden. In diesem Handbuch werden einige der gebräuchlichsten Anwendungsfälle untersucht, wobei der Schwerpunkt auf der Integration mit [jQuery](https://jquery.com/) und [Backbone](https://backbonejs.org/) liegt wird angewendet, um Komponenten in vorhandenen Code zu integrieren.

Reviewer: > In diesem Handbuch werden einige der gebräuchlichsten Anwendungsfälle untersucht, wobei der Schwerpunkt auf der Integration mit [jQuery](https://jquery.com/) und [Backbone](https://backbonejs.org/) **liegt wird angewendet, um Komponenten in vorhandenen Code zu integrieren.**

The highlighted part is kinda weird, maybe something like:

> In diesem Handbuch werden einige der gebräuchlichsten Anwendungsfälle untersucht, wobei der Schwerpunkt auf der Integration mit [jQuery](https://jquery.com/) und [Backbone](https://backbonejs.org/) liegt, der gleiche Ansatz kann jedoch angewandt werden, um Komponenten in beliebigen, bereits vorhandenen Code zu integrieren.

## Integration mit DOM-Manipulations-Plugins {#integrating-with-dom-manipulation-plugins}

React kennt keine Änderungen am DOM außerhalb von React. Es ermittelt Aktualisierungen auf der Grundlage seiner eigenen internen Darstellung. Wenn dieselben DOM-Knoten von einer anderen Bibliothek bearbeitet werden, ist React verwirrt und kann nicht wiederhergestellt werden.

Dies bedeutet nicht, dass es unmöglich oder sogar notwendig ist, React mit anderen Möglichkeiten zur Beeinflussung des DOM zu kombinieren. Du musst lediglich bedenken, was jeder tut.

Der einfachste Weg, Konflikte zu vermeiden, besteht darin, zu verhindern, dass die React-Komponente aktualisiert wird. Du kannst dies tun, indem du Elemente renderst, für die React keinen Grund zum Aktualisieren hat, z. B. ein leeres `<div />`.

### Wie nähere du dich dem Problem {#how-to-approach-the-problem}

Um dies zu demonstrieren, skizzieren wir einen Wrapper für ein generisches jQuery-Plugin.

Wir werden eine [ref](/ docs / refs-and-the-dom.html) an das Root-DOM-Element anhängen. In `componentDidMount` erhalten wir einen Verweis darauf, damit wir ihn an das jQuery-Plugin übergeben können.

> Um zu verhindern, dass React das DOM nach dem Mounten berührt, geben wir ein leeres `<div />` von der `render()`-Methode zurück. Das `<div />`-Element hat keine Eigenschaften oder untergeordneten Elemente, daher hat React keinen Grund, es zu aktualisieren, sodass das jQuery-Plugin diesen Teil des DOM verwalten kann:

Reviewer: ```suggestion
Um zu verhindern, dass React das DOM nach dem Mounten berücksichtigt, geben wir ein leeres `<div />` von der `render()`-Methode zurück. Das `<div />`-Element hat keine Eigenschaften oder untergeordneten Elemente, daher hat React keinen Grund, es zu aktualisieren, sodass das jQuery-Plugin diesen Teil des DOM verwalten kann:
```

}
```

> Beachte, dass wir sowohl `componentDidMount` als auch `componentWillUnmount` [Lebenszyklusmethoden](/docs/react-component.html#the-component-lifecycle) definiert haben. Viele jQuery-Plugins hängen Event-Listener an das DOM an, daher ist es wichtig, sie in "componentWillUnmount" zu entfernen. Wenn das Plugin keine Bereinigungsmethode bereitstellt, musst du wahrscheinlich eine eigene Methode bereitstellen. Denke daran, alle vom Plugin registrierten Event-Listener zu entfernen, um Speicherlecks zu vermeiden.

Reviewer: ```suggestion
Beachte, dass wir sowohl `componentDidMount` als auch `componentWillUnmount` [Lebenszyklusmethoden](/docs/react-component.html#the-component-lifecycle) definiert haben. Viele jQuery-Plugins hängen Event-Listener an das DOM an, daher ist es wichtig, sie in `componentWillUnmount` zu entfernen. Wenn das Plugin keine Bereinigungsmethode bereitstellt, musst du wahrscheinlich eine eigene Methode bereitstellen. Denke daran, alle vom Plugin registrierten Event-Listener zu entfernen, um Speicherlecks zu vermeiden.
```

Author: I don't understand why you want to change this function reference to be in quotes.

Reviewer: I want to change it back to backticks, since it was surrounded with quotes in your translation.

Author: 🤦Read it backwards.

### Integration mit dem jQuery Chosen-Plugin {#integrating-with-jquery-chosen-plugin}

Um ein konkreteres Beispiel für diese Konzepte zu finden, schreiben wir einen minimalen Wrapper für das Plugin [Chosen](https://harvesthq.github.io/chosen/), der die Eingaben von `<select>` erweitert.

>**Hinweis:**
>

> >Nur weil es möglich ist, heißt das nicht, dass es der beste Ansatz für React-Apps ist. Wir empfehlen deinen, React-Komponenten zu verwenden, wenn du kannst. React-Komponenten lassen sich in React-Anwendungen leichter wiederverwenden und bieten häufig eine bessere Kontrolle über ihr Verhalten und Erscheinungsbild.

Reviewer: ```suggestion
>Nur weil es möglich ist, heißt das nicht, dass es der beste Ansatz für React-Apps ist. Wir empfehlen dir die Verwendung von React-Komponenten, wenn du die freie Wahl hast. React-Komponenten lassen sich in React-Anwendungen leichter wiederverwenden und bieten häufig eine bessere Kontrolle über ihr Verhalten und Erscheinungsbild.
```

}
```

> Beachte, wie wir `<select>` in ein zusätzliches `<div>` eingewickelt haben. Dies ist notwendig, da Chosen ein weiteres DOM-Element direkt nach dem Knoten `<select>` anfügt, den wir an ihn übergeben haben. In Bezug auf React hat `<div>` jedoch immer nur ein einziges untergeordnetes Element. Auf diese Weise stellen wir sicher, dass React-Updates nicht mit dem von Chosen angehängten zusätzlichen DOM-Knoten in Konflikt stehen. Wenn du das DOM außerhalb von React Flow änderst, musst du sicherstellen, dass React keinen Grund hat, diese DOM-Knoten zu berühren.

Reviewer: ```suggestion
Beachte, wie wir `<select>` mit einem zusätzlichen `<div>` umschlossen haben. Dies ist notwendig, da Chosen ein weiteres DOM-Element direkt nach dem Knoten `<select>` anfügt, den wir an ihn übergeben haben. In Bezug auf React hat `<div>` jedoch immer nur ein einziges untergeordnetes Element. Auf diese Weise stellen wir sicher, dass React-Updates nicht mit dem von Chosen angehängten zusätzlichen DOM-Knoten in Konflikt stehen. Wenn du das DOM außerhalb von React Flow änderst, musst du sicherstellen, dass React keinen Grund hat, diese DOM-Knoten zu berücksichtigen.
```

> Als nächstes werden wir die Lebenszyklusmethoden implementieren. Wir müssen Chosen mit dem ref auf den `<select>`-Knoten in `componentDidMount` initialisieren und ihn in `componentWillUnmount` herunterreißen:

Reviewer: ```suggestion
Als nächstes werden wir die Lebenszyklusmethoden implementieren. Wir müssen Chosen mit dem ref auf den `<select>`-Knoten in `componentDidMount` initialisieren und ihn in `componentWillUnmount` zerstören:
```

Not sure if this fits better than herunterreißen

}
```

[**Probiere es mit CodePen aus**](https://codepen.io/gaearon/pen/qmqeQx?editors=0010)

Beachte, dass React dem Feld `this.el` keine besondere Bedeutung zuweist. Es funktioniert nur, weil wir dieses Feld zuvor von einem `ref` in der `render()`-Methode zugewiesen haben:

```js
<select className="Chosen-select" ref={el => this.el = el}>
```

> Dies reicht aus, um unsere Komponente rendern zu lassen, wir möchten jedoch auch über die Wertänderungen informiert werden. Dazu abonnieren wir das jQuery-Events `change` für das von Chosen verwaltete `<select>`.

Reviewer: ```suggestion
Dies reicht aus, um unsere Komponente rendern zu lassen, wir möchten jedoch auch über die Wertänderungen informiert werden. Dazu abonnieren wir das `change` jQuery-Event für das von Chosen verwaltete `<select>`.
```

> Wir werden `this.props.onChange` nicht direkt an Chosen übergeben, da sich die Requisiten der Komponente im Laufe der Zeit ändern können, und dies schließt Event-Handler ein. Stattdessen deklarieren wir eine `handleChange()`-Methode, die `this.props.onChange` aufruft und das jQuery-Events `change` abonniert:

Reviewer: ```suggestion
Wir werden `this.props.onChange` nicht direkt an Chosen übergeben, da sich die Eigenschaften der Komponente im Laufe der Zeit ändern können, und dies schließt Event-Handler ein. Stattdessen deklarieren wir eine `handleChange()`-Methode, die `this.props.onChange` aufruft und abonnieren damit das `change` jQuery-Event:
```

}
```

[**Probiere es mit CodePen aus**](https://codepen.io/gaearon/pen/bWgbeE?editors=0010)

> Schließlich bleibt noch etwas zu tun. In React können sich Requisiten im Laufe der Zeit ändern. Beispielsweise kann die Komponente `<Chosen>` verschiedene untergeordnete Elemente erhalten, wenn sich der State der übergeordneten Komponente ändert. Dies bedeutet, dass es an Integrationspunkten wichtig ist, dass wir das DOM als Reaktion auf Requisitenaktualisierungen manuell aktualisieren, da React das DOM nicht mehr für uns verwalten kann.

Reviewer: ```suggestion
Abschließend müssen wir noch eine Sache erledigen. In React können sich Eigenschaften im Laufe der Zeit ändern. Beispielsweise kann die Komponente `<Chosen>` verschiedene untergeordnete Elemente erhalten, wenn sich der State der übergeordneten Komponente ändert. Dies bedeutet, dass es an Integrationspunkten wichtig ist, dass wir das DOM manuell aktualisieren, um auf Änderungen der Eigenschaften zu reagieren, da React das DOM nicht mehr für uns verwalten kann.
```

> Aus der Dokumentation von Chosen geht hervor, dass wir die jQuery `trigger()`-API verwenden können, um Änderungen am ursprünglichen DOM-Element zu melden. Wir überlassen es React, `this.props.children` in` <select> `zu aktualisieren, fügen aber auch eine `componentDidUpdate()`-Lebenszyklusmethode hinzu, die Chosen über Änderungen in der Liste der untergeordneten Elemente benachrichtigt:

Reviewer: ```suggestion
Aus der Dokumentation von Chosen geht hervor, dass wir die jQuery `trigger()`-API verwenden können, um Änderungen am ursprünglichen DOM-Element zu melden. Wir überlassen es React, `this.props.children` in `<select>` zu aktualisieren, fügen aber auch eine `componentDidUpdate()`-Lebenszyklusmethode hinzu, die Chosen über Änderungen in der Liste der untergeordneten Elemente benachrichtigt:
```

}
```

> Auf diese Weise kann Chosen das DOM-Element aktualisieren, wenn sich die von React verwalteten `<select>` untergeordneten Elemente ändern.

Reviewer: ```suggestion
Auf diese Weise kann Chosen das DOM-Element aktualisieren, wenn sich die von React verwalteten, dem `<select>` untergeordneten Elemente ändern.
```
